### DISCUSSION TITLE
minus / subtract in automation throwing error

### DISCUSSION THREAD

**Discussion by MrWombastic** [Budibase/budibase#8082] [Help]
Hi,

I got a formula where you can create a relationship between a Sell-Transaction and a Buy-Transaction (Table: Applicationsheet).
![image](https://user-images.githubusercontent.com/34077172/193407565-0bc42b24-6dc1-4b66-980c-9385d483bc00.png)

The Transactions are coming from a table, where Sell- and Buy- Transactions can be inserted with a Quantity, an open Quantity, a Value, a Price, and same other data about the transaction. In this Table, I want to update the OpenQuantity Field when the formula above gets saved.
![image](https://user-images.githubusercontent.com/34077172/193407745-974892c5-ad5e-4aeb-afbf-e9afa1a6a10f.png)

So the process looks like this:
1. Create the Sell-Transaction
2. Use the Formula to connect the sell transaction to 1 ore more buy transactions
3. AUTMATION: Update the Open Quantity in the Transactiontable

I am failing with the automation.

The Process inside the automation is:
1. Trigger Insert Row in the Table, where transactions are connected
2. Query rows inside the Transactiontable filtered by the Buy-Transaction-ID's entered in the Trigger-Table
3. Update row, filtered by the Buy-Transaction-ID from the Trigger-Event and with the Calculation steps.1.rows.OpenQuantity - Trigger.row.Quantity

![image](https://user-images.githubusercontent.com/34077172/193407850-0d92b969-89e7-4ef4-8bbf-1f0c1b11692e.png)
Filter in the Query (to get the OpenQuantity of the row)
![image](https://user-images.githubusercontent.com/34077172/193407853-46694c46-a6e6-4dfd-9045-f8594afcb0b1.png)
The Update Row Event
![image](https://user-images.githubusercontent.com/34077172/193407924-e11c09b5-1c93-4b08-94e4-b6e2f79dd303.png)
![image](https://user-images.githubusercontent.com/34077172/193407917-f10f04a6-ce0f-42d6-9c54-2904bcc11855.png)
![image](https://user-images.githubusercontent.com/34077172/193407907-02ce25dd-0e3d-4641-93ae-9d8c63955ec1.png)

The result looks like this:
![image](https://user-images.githubusercontent.com/34077172/193407390-3b65fb4d-c9dd-4b9b-a06e-4ceac1173422.png)

I tried the same automation with both fields in the subtract calculation, so simply updating the field with it's own value or with the value from the trigger, both without subtract, this works. I also tried the subtract and minus helper together with conversion helpers like toInt or toFloat before the subtract but this also doesn't work.

I searched over all docs / disucssions but didn't find a solution.

And if we find the problem, I'd like to ask if it's possible to work with multiple Buy-Transactions in one step. Because in the Formula you can choose multiple Buy-Transactions with one Sell-Transaction. So you can buy 2 times 5 of something and sell all 10 at one time. So I'd like to get the OpenQuantity Column set to zero for all three entries, the Sell-Transaction and both Buy-Transactions.

**Comment by melohagan** [comment]
Hey @MrWombastic 

Just to clarify, are you speaking about Budibase DB formulas? https://docs.budibase.com/docs/formula
Would you be able to show my what your formula looks like if that is the case?
